packages/redis/src/RedisClient.ts

```diff
@@ -214,4 +264,260 @@ export class RedisClient implements IRedisClient, IRequireInitialization, IDispo
+        } catch (err) {
+            const alreadyExistsErrorMessage = "BUSYGROUP Consumer Group name already exists";
+            if (supressAlreadyExistsError && err.message === alreadyExistsErrorMessage) {
+                // TODO should this be considered an error or success?
+                this.metrics.increment(RedisMetrics.XGroupAlreadyExists, {
+                    db,
+                    streamName,
+                    consumerGroup,
+                    result: RedisMetricResults.Success,
+                });
+
+                return "OK";
+            }
+
+            failSpan(span, err);
+            this.metrics.increment(RedisMetrics.XGroupCreate, {
+                db,
+                streamName,
+                consumerGroup,
+                result: RedisMetricResults.Error,
+                error: err,
+            });
+
+            throw err;
+        } finally {
+            span.finish();
+        }
+    }
+
+    public async xAck(
+        context: SpanContext,
+        streamName: string,
+        consumerGroup: string,
+        streamId: string
+    ): Promise<number> {
+        const db = this.config.db;
+        const span = this.tracer.startSpan("Redis Client xAck Call", { childOf: context });
+        this.spanLogAndSetTags(span, this.xAck.name, this.config.db, undefined, streamName);
+        try {
+            const response = await this.client.xack(streamName, consumerGroup, streamId);
+            this.metrics.increment(RedisMetrics.XAck, {
+                db,
+                streamName,
+                consumerGroup,
+                result: RedisMetricResults.Success,
+            });
+            return response;
+        } catch (err) {
+            failSpan(span, err);
+            this.metrics.increment(RedisMetrics.XAck, {
+                db,
+                streamName,
+                consumerGroup,
+                result: RedisMetricResults.Error,
+                error: err,
+            });
+
+            throw err;
+        } finally {
+            span.finish();
+        }
+    }
+
+    public async xReadGroup(
+        context: SpanContext,
+        streamName: string,
+        consumerGroup: string,
+        consumerName: string,
+        count: number,
+        id: string = ">"
+    ): Promise<IRedisMessage[]> {
+        const db = this.config.db;
+        const span = this.tracer.startSpan("Redis Client xReadGroupObject Call", {
+            childOf: context,
+        });
+
+        this.spanLogAndSetTags(span, this.xReadGroup.name, this.config.db, id, streamName);
+
+        try {
+            const response = await this.client.xreadgroup([
+                "group",
+                consumerGroup,
+                consumerName,
+                "count",
+                String(count),
+                "block",
+                "0",
+                "streams",
+                streamName,
+                id,
+            ]);
+
+            console.log("HERE YO: ", response);
```

Comment:
nit: you'll wanna remove this console.log

Reply:
Good catch haha